### PR TITLE
Added google map in event detail page

### PR DESCRIPTION
**Pull request by mlCode111**
- added google maps in event detail page (see attached screenshot). File updated: get_together/templates/get_together/events/show_event.html 
Added map

- refactored google maps API code to accurately locate the address of the place (see attached screenshot). The original code was not able to accurately locate places that are not public or have multiple locations. File updated: get_together/templates/get_together/events/show_event.html
Refactored google maps api query

### CODE REVIEW

File: get_together/templates/get_together/events/show_event.html [hunk 1]
```diff
@@ -202,6 +202,19 @@ <h2>{{ event.name }}</h2>
             </div>
             {% endif %}
 
+            <div class="row mb-2">
+              <div class="col-12">
+                <iframe
+                  width="100%"
+                  height="300"
+                  frameborder="0" style="border:0"
+                  src="https://www.google.com/maps/embed/v1/place?key={{ settings.GOOGLE_MAPS_API_KEY }}&q={{event.place.address|urlencode}}%2C%20{{event.place.city|urlencode}}&zoom=14"
```
Comment (thiagoferreiraw) [inline get_together/templates/get_together/events/show_event.html:212]: So, we are using the place address and city to filter the place on google maps. I'm not sure if this is a good approach. Currently, the places can be inputted by the user, and there's no guarantee the entered address is correct and can be found in google maps.

@pawalls , any idea here? In the future, It'll be interesting to implement a smooth integration with google maps. Do you think we can use the address for now?

**Comment by thiagoferreiraw** [comment]
There is an option to host events without a place; In those cases, we must check if the place exists before showing the map. Otherwise, it looks like this:

image

Can you please add this condition @mdigitalcomm ? Thank you!

File: get_together/templates/get_together/events/show_event.html (hunk 1 above)
Comment (mlCode111) [reply get_together/templates/get_together/events/show_event.html:212]: Originally the app used the name of the place and city to filter the place.  The upside is the place's name can show in the map, if the place on Google is exactly the one the user is looking for. Sometimes there are discrepancies, such as when the place is not in Google's Place library. That's why I chose to use the address and city. I'd love to hear your suggestions.

Comment (pawalls-zz) [reply get_together/templates/get_together/events/show_event.html:212]: @thiagoferreiraw -- would your concerns be alleviated if we used Google Map search for selecting location during event creation? I believe we'd end up with a lat/long or place id in the database and then would use the maps API to translate that back to a place.

**Review by pawalls-zz — CHANGES_REQUESTED** [review]
As written, this is unlikely to be accepted upstream. A simple fix would be to wrap the entire block with `{% if settings.GOOGLE_MAPS_API_KEY %}` so that the map is only shown when an API key is provided.

File: get_together/templates/get_together/events/show_event.html (hunk 1 above)
Comment (thiagoferreiraw) [reply get_together/templates/get_together/events/show_event.html:212]: yes @pawalls! Seems much less likely to fail. Can we log an issue and add to the board for this?

Comment (pawalls-zz) [reply get_together/templates/get_together/events/show_event.html:212]: @thiagoferreiraw -- done.

**Review by thiagoferreiraw — APPROVED** [review]
well done @mdigitalcomm ! Thank you very much🙏

**Comment by pawalls-zz** [comment]
Looks good to me!

**Pull request merged**